src/derivkit/forecasting/calculus.py

```diff
@@ -281,3 +320,62 @@ def _check_scalar_valued(function, theta0: np.ndarray, i: int, n_workers: int):
 def gauss_newton_hessian(*args, **kwargs):
     """This is a placeholder for a Gauss-Newton Hessian computation function."""
     raise NotImplementedError
+
+
+def _grad_wrt_param(
+    function: Callable[[ArrayLike], ArrayLike | float],
+    theta0: ArrayLike,
+    idx: int,
+) -> NDArray[np.floating]:
+    """Derivative of a vector-valued function wrt a single parameter theta[idx].
+
+     Helper used by ``jacobian``. Wraps ``function`` into a single-variable callable via
+    ``derivkit.utils.get_partial_function`` and differentiates it with
+    ``DerivativeKit.adaptive.differentiate``.
+
+    Args:
+        function: The vector-valued function to be differentiated.
+        theta0: The parameter vector at which the derivative is evaluated.
+        idx: Zero-based index of the parameter with respect to which to differentiate.
+
+    Returns:
+        A 1D array representing the derivative of the function with respect to theta[idx].
+    """
+    theta0_x = deepcopy(np.atleast_1d(theta0))
+    f_i = get_partial_function(function, idx, theta0_x)   # this sets theta[idx]=y
+    kit = DerivativeKit(f_i, theta0_x[idx])
+    # Keep inner serial to avoid nested pools; adaptive can still batch-eval internally.
+    inner_workers = 1
+    gi = kit.adaptive.differentiate(order=1, n_workers=inner_workers)
```

Comment:
```suggestion
    gi = kit.adaptive.differentiate(order=1)
```
Since the default behaviour of the `adaptive` method is to not use multiprocessing.